### PR TITLE
fix(fa): fixing spouse personal tax calculation 

### PR DESCRIPTION
**Pull request by MargretFinnboga**
# ...

[Skoða upphæðir þegar persónuafsláttur maka ](https://app.asana.com/0/1206695048257743/1206286875981909)

## What

- Showing tax according to personal tax 
- Fixing calculation spouse personal tax is added 

## Why

- So the calculations are right and more clear for staff 

## Screenshots / Gifs
BEFORE
Screenshot 2024-06-05 at 13 03 19

AFTER
Screenshot 2024-06-05 at 14 34 40
Screenshot 2024-06-05 at 14 34 33Screenshot 2024-06-05 at 14 34 25

## Checklist:

- [x] I have performed a self-review of my own code
- [ ] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [x] Formatting passes locally with my changes
- [x] I have rebased against main before asking for a review

## Summary by CodeRabbit

- **New Features**
	- Improved calculation of final tax amounts and allowances in financial aid applications.
- **Refactor**
	- Enhanced logic for determining final tax amounts, ensuring allowances do not exceed calculated tax.
- **Tests**
	- Added tests for the new `calculateFinalTaxAmount` function to cover various scenarios involving personal and spouse credits.

### CODE REVIEW

File: apps/financial-aid/web-veita/src/components/ModalTypes/AcceptModal.tsx [hunk 1]
```diff
@@ -310,11 +314,12 @@ const AcceptModal = ({
           value={Number(state.personalTaxCreditPercentage).toString()}
           type="number"
```
Comment (Rikkiabb) [inline apps/financial-aid/web-veita/src/components/ModalTypes/AcceptModal.tsx:315]: Should persónuafsláttur be postfixed with '%' or is it clear that this is always a percentage? I wrote a helper hook the other day that did exactly that. Holla if you want to use it 😱

File: libs/financial-aid/shared/src/lib/taxCalculator.spec.ts [hunk 2]
```diff
@@ -130,30 +131,44 @@ describe('Tax calculator', () => {
       const acceptedAidFinalAmount = calculateAcceptedAidFinalAmount(
         100000,
         100,
-        100,
       )
-      expect(acceptedAidFinalAmount).toEqual(68520)
+      expect(acceptedAidFinalAmount).toEqual(99900)
     })
 
     test('should return accepted final amount with 10% personal and 30% spose credit', () => {
```
Comment (Rikkiabb) [inline libs/financial-aid/shared/src/lib/taxCalculator.spec.ts:138]: *spouse

File: libs/financial-aid/shared/src/lib/taxCalculator.spec.ts [hunk 3]
```diff
@@ -130,30 +131,44 @@ describe('Tax calculator', () => {
...
-        10,
-        30,
+        300000,
+        64926,
       )
-      expect(acceptedAidFinalAmount).toEqual(55535)
+      expect(acceptedAidFinalAmount).toEqual(235074)
     })
 
     test('should return accepted final amount with 10% personal and 30% spose credit', () => {
```
Comment (Rikkiabb) [inline libs/financial-aid/shared/src/lib/taxCalculator.spec.ts:146]: *spouse

**Review by Rikkiabb — APPROVED** [review]
Nice just a tinkering about the design and nits on old code.

File: apps/financial-aid/web-veita/src/components/ModalTypes/AcceptModal.tsx (hunk 1 above)
Comment (MargretFinnboga) [reply apps/financial-aid/web-veita/src/components/ModalTypes/AcceptModal.tsx:315]: I am going to make this nice have and add it to the backlog

**Pull request merged**